drivers/regulator/regulator_npm1100.c

```diff
@@ -0,0 +1,262 @@
+		return -EINVAL;
+	}
+
+	return linear_range_group_get_value(
+		buck_ranges, ARRAY_SIZE(buck_ranges), idx, volt_uv);
+}
+
+static int regulator_npm1100_set_voltage(const struct device *dev,
+					 int32_t min_uv, int32_t max_uv)
+{
+	const struct regulator_npm1100_config *config = dev->config;
+	uint16_t idx;
+	int ret;
+
+	if ((config->voutbset0.port == NULL) ||
+	    (config->voutbset1.port == NULL)) {
+		return -ENOTSUP;
+	}
+
+	ret = linear_range_group_get_win_index(
+		buck_ranges, ARRAY_SIZE(buck_ranges), min_uv, max_uv, &idx);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = gpio_pin_set_dt(&config->voutbset0, !!(idx & BIT(0)));
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = gpio_pin_set_dt(&config->voutbset1, !!(idx & BIT(1)));
+	if (ret < 0) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int regulator_npm1100_get_voltage(const struct device *dev,
+					 int32_t *volt_uv)
+{
+	const struct regulator_npm1100_config *config = dev->config;
+	uint16_t idx;
+	int ret;
+
+	if ((config->voutbset0.port == NULL) ||
+	    (config->voutbset1.port == NULL)) {
+		return -ENOTSUP;
+	}
+
+	ret = gpio_pin_get_dt(&config->voutbset0);
+	if (ret < 0) {
+		return ret;
+	}
+
+	idx = (uint8_t)ret;
+
+	ret = gpio_pin_get_dt(&config->voutbset1);
+	if (ret < 0) {
+		return ret;
+	}
+
+	idx |= (uint8_t)ret << 1U;
+
+	return linear_range_group_get_value(
+		buck_ranges, ARRAY_SIZE(buck_ranges), idx, volt_uv);
+}
+
+static int regulator_npm1100_set_mode(const struct device *dev,
+				      regulator_mode_t mode)
+{
+	const struct regulator_npm1100_config *config = dev->config;
+
+	if ((config->mode.port == NULL) || (mode > NPM1100_MODE_PWM)) {
+		return -ENOTSUP;
+	}
+
+	return gpio_pin_set_dt(&config->mode,
+			       mode == NPM1100_MODE_AUTO ? 0 : 1);
+}
+
+static int regulator_npm1100_get_mode(const struct device *dev,
+				      regulator_mode_t *mode)
+{
+	const struct regulator_npm1100_config *config = dev->config;
+	int ret;
+
+	if (config->mode.port == NULL) {
+		return -ENOTSUP;
+	}
+
+	ret = gpio_pin_get_dt(&config->mode);
+	if (ret < 0) {
+		return ret;
+	}
+
+	*mode = (ret == 0) ? NPM1100_MODE_AUTO : NPM1100_MODE_PWM;
+
+	return 0;
+}
+
+static int regulator_npm1100_init(const struct device *dev)
+{
+	const struct regulator_npm1100_config *config = dev->config;
+	int ret;
+
+	if (config->mode.port != NULL) {
+		if (!gpio_is_ready_dt(&config->mode)) {
+			return -ENODEV;
+		}
+
+		ret = gpio_pin_configure_dt(&config->mode,
+					    GPIO_INPUT | GPIO_OUTPUT_INACTIVE);
+		if (ret < 0) {
+			return ret;
+		}
+	}
+
+	if ((config->voutbset0.port != NULL) ||
+	    (config->voutbset1.port != NULL)) {
+		if (!gpio_is_ready_dt(&config->voutbset0) ||
+		    !gpio_is_ready_dt(&config->voutbset1)) {
+			return -ENODEV;
+		}
+
+		ret = gpio_pin_configure_dt(&config->voutbset0,
+					    GPIO_INPUT | GPIO_OUTPUT_INACTIVE);
+		if (ret < 0) {
+			return ret;
+		}
+
+		ret = gpio_pin_configure_dt(&config->voutbset1,
+					    GPIO_INPUT | GPIO_OUTPUT_INACTIVE);
```

Comment:
This will always configure the BUCK output voltage to 1.8 V initially. Shouldn't `config->common.min_uv` be taken into account here?

Reply:
Ok, I actually realized this driver had a big problem. Even if it seems to work, the spec says:

> BUCK output voltage selection pins VOUTBSET0 and VOUTBSET1 should be hardwired to DEC, VSYS, or AVSS. Do not toggle these pins during operation.

So I've removed the capability of configuring voltage entirely. My fault reading specs too fast.